backend/tests/apps/api/rest/v0/pagination_test.py

```diff
@@ -0,0 +1,293 @@
+
+        assert result["current_page"] == 1
+        assert result["total_count"] == 25
+        assert result["total_pages"] == 3
+        assert result["has_next"] is True
+        assert result["has_previous"] is False
+        assert result["items"] == ["item1", "item2", "item3"]
+        mock_queryset.__getitem__.assert_called_once_with(slice(0, 10))
+
+    def test_paginate_middle_page(self):
+        """Test pagination on a middle page."""
+        paginator = CustomPagination()
+
+        mock_queryset = MagicMock()
+        mock_queryset.count.return_value = 35
+        mock_queryset.__getitem__ = MagicMock(return_value=["item11", "item12"])
+
+        mock_pagination = MagicMock()
+        mock_pagination.page = 2
+        mock_pagination.page_size = 10
+
+        result = paginator.paginate_queryset(mock_queryset, mock_pagination)
+
+        assert result["current_page"] == 2
+        assert result["total_count"] == 35
+        assert result["total_pages"] == 4
+        assert result["has_next"] is True
+        assert result["has_previous"] is True
+        assert result["items"] == ["item11", "item12"]
+        mock_queryset.__getitem__.assert_called_once_with(slice(10, 20))
+
+    def test_paginate_last_page(self):
+        """Test pagination on the last page."""
+        paginator = CustomPagination()
+
+        mock_queryset = MagicMock()
+        mock_queryset.count.return_value = 25
+        mock_queryset.__getitem__ = MagicMock(
+            return_value=["item21", "item22", "item23", "item24", "item25"]
+        )
+
+        mock_pagination = MagicMock()
+        mock_pagination.page = 3
+        mock_pagination.page_size = 10
+
+        result = paginator.paginate_queryset(mock_queryset, mock_pagination)
+
+        assert result["current_page"] == 3
+        assert result["total_count"] == 25
+        assert result["total_pages"] == 3
+        assert result["has_next"] is False
+        assert result["has_previous"] is True
+        assert result["items"] == ["item21", "item22", "item23", "item24", "item25"]
+        mock_queryset.__getitem__.assert_called_once_with(slice(20, 30))
+
+    def test_paginate_single_page(self):
+        """Test pagination when all items fit on a single page."""
+        paginator = CustomPagination()
+
+        mock_queryset = MagicMock()
+        mock_queryset.count.return_value = 5
+        mock_queryset.__getitem__ = MagicMock(
+            return_value=["item1", "item2", "item3", "item4", "item5"]
+        )
+
+        mock_pagination = MagicMock()
+        mock_pagination.page = 1
+        mock_pagination.page_size = 10
+
+        result = paginator.paginate_queryset(mock_queryset, mock_pagination)
+
+        assert result["current_page"] == 1
+        assert result["total_count"] == 5
+        assert result["total_pages"] == 1
+        assert result["has_next"] is False
+        assert result["has_previous"] is False
```

Comment:
I'm not sure if there is a way to enforce it -- we don't use `is` for boolean assertions.
```suggestion
        assert not result["has_previous"]
```